Arelle/Arelle
# Add implementation of XPath function fn:tokenize (for formula linkbase to use)

#### Reason for change
Arelle is missing support for some XPath 2.0 standard functions and there is a good chance that an upcoming taxonomy might use [`fn:tokenize()`](https://www.w3.org/TR/xpath-functions/#func-tokenize) so add an implementation of it. 

#### Description of change
Implement `fn:tokenize()` and add a few tests for it.

#### Steps to Test
A simple formula linkbase making use of tokenize (zipped purely to get past Github upload filter,)
[test-tokenize-formula.zip](https://github.com/user-attachments/files/16175679/test-tokenize-formula.zip)


```
python arelleCmdLine.py -v -f test-tokenize-formula.xml
```

Without this pull request it fails:
```
[info] loaded in 0.03 secs at 2024-07-11T11:48:35 - C:\b\test-tokenize-formula.xml
[] Formula xpath2 grammar initialized in 0.63 secs -
[err:XPST0017] Variable set AssertionTokenizeAndJoin
Exception: Function named fn:tokenize does not have a custom or built-in implementation. - test-tokenize-formula.xml 14
[err:XPST0017] Variable set AssertionCountTokensZeroMatch
Exception: Function named fn:tokenize does not have a custom or built-in implementation. - test-tokenize-formula.xml 11
[err:XPST0017] Variable set AssertionCountTokens
Exception: Function named fn:tokenize does not have a custom or built-in implementation. - test-tokenize-formula.xml 8
[info] validated in 0.66 secs - C:\b\test-tokenize-formula.xml
```

With this pull request it succeeds:
```
[info] loaded in 0.03 secs at 2024-07-11T11:53:45 - C:\b\test-tokenize-formula.xml
[] Formula xpath2 grammar initialized in 1.01 secs -
[message:AssertionTokenizeAndJoin] Turn '  a b  c   ' into 'abc' 'abc'  - test-tokenize-formula.xml 14
[message:AssertionCountTokens] Length of count(fn:tokenize('a-b-c-d-e-f-g', '-')) should be 7 and actually is '7'  - test-tokenize-formula.xml 8
[message:AssertionCountTokensZeroMatch] Length of count(fn:tokenize('a-b-c-d-e-f-g-', '-')) should be 8 and actually is '8'  - test-tokenize-formula.xml 11
[info] validated in 1.05 secs - C:\b\test-tokenize-formula.xml
```

Alternately, the unit tests can be run:
```
pytest .\tests\unit_tests\arelle\test_functionfn.py
```

**review**:
@Arelle/arelle
